src/queue.rs

```diff
@@ -135,6 +214,20 @@ impl Queue {
     }
 }
 
+impl Drop for Queue {
```

Comment:
This raises an important point - where is this getting dropped? It shouldn't be getting dropped until processes exit. Also it should nack on drop if anything. Though ideally it wouldn't store handles in the structure at all.